packages/instantsearch.js/src/lib/__tests__/server.test.ts

```diff
@@ -421,4 +421,76 @@ describe('getInitialResults', () => {
       ]
     `);
   });
+
+  test('injects clickAnalytics and userToken into the state', async () => {
+    const search = instantsearch({
+      indexName: 'indexName',
+      searchClient: createSearchClient(),
+      initialUiState: {
+        indexName: {
+          query: 'apple',
+        },
+      },
+      insights: true,
+    });
+
+    search.addWidgets([connectSearchBox(() => {})({})]);
+
+    search.start();
+
+    const requestParams = await waitForResults(search);
+
+    expect(requestParams).toEqual([
+      {
+        clickAnalytics: true,
+        query: 'apple',
+        userToken: expect.stringMatching(/^anonymous-/),
+      },
+    ]);
+
+    const initialResults = getInitialResults(search.mainIndex, requestParams);
+    expect(initialResults).toEqual({
+      indexName: {
+        requestParams: [
+          {
+            clickAnalytics: true,
+            query: 'apple',
+            userToken: expect.stringMatching(/^anonymous-/),
```

Comment:
what do you mean, `toBeDefined`? I'd like to assert that it has a reasonable value too. Or do you mean two assertions, one for `clickAnalytics).toBe(true)` and one for userToken to avoid the objectContaining?